js/shared.js

```diff
@@ -374,3 +376,160 @@ async function findMatchingIncipits(incipit, titleStart) {
             return Promise.resolve([])
         })
 }
+
+/**
+ * @param source A String that is either a text body or a URI to a text resource.
+ */ 
+async function getAllWitnessesOfSource(source){
+    const linesLoaded = document.querySelector("tpen-line-selector").hasAttribute("tpen-lines-loaded") ? true : false
+    if(!linesLoaded){
+        return Promise.reject("There is no reason to run this function because we cannot supply the results to a non-existent UI.  Wait for the T-PEN Transcription to load.")
+    }
+    // Other asyncronous loading functionality may have already built this.  Use what is cached if so.
+    if(Object.keys(witnessesObj).length > 0){
+        for(const witnessURI in witnessesObj){
```

Comment:
The conditional is not needed because the for loop will just short out if the length is 0 or if the object is undefined.

Reply:
Yes but I do the `return` in this scenario anyway.  The loop could be before this and it could be followed by the single-line-if.